app/uk/gov/hmrc/helptosavereminder/actors/TestOnlyActor.scala

```diff
@@ -68,6 +68,7 @@ class TestOnlyActor(repository: HtsReminderMongoRepository)(implicit ec: Executi
         dateAcknowledged.map(_.toString).orNull
       )
     case SendEmails(emails) =>
+      implicit val ec: ExecutionContext = context.dispatcher
```

Comment:
As this action involves making DB calls, generally not recommended to use context dispatcher. Because, unless otherwise there is an explicit dispatcher created for this send-email, it would default to system dispatcher. And, using that for either blocking or long running futures (especially if those futures are DB related actions), context dispatcher would starve other actors for resources.

For this case, I would recommend recommend creating a separate, dedicated, dispatcher and then using that instead to isolate the workloads for this actor.

Reply:
Added a local execution context.